test/unit/platform.test.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import * as platform from '../../src/platform';
+import * as logging from '../../src/logging';
+import fs from 'fs';
+import sinon from 'sinon';
+
+// Mock the getPlatform function
```

Comment:
NP: I think this comment can be removed 

```suggestion
```

Reply:
thanks, I actually had a getPlatform method that I cleaned up when I decided to go with sinon for mocking.